src/XRP/core-xrpl-client.ts

```diff
@@ -313,15 +316,19 @@ export default class CoreXrplClient implements CoreXrplClientInterface {
     transactionHash: string,
     wallet: Wallet,
   ): Promise<TransactionResult> {
-    const rawTransactionStatus = await this.waitForFinalTransactionOutcome(
-      transactionHash,
-      wallet,
+    const {
+      rawTransactionStatus,
+      lastLedgerPassed,
+    } = await this.waitForFinalTransactionOutcome(transactionHash, wallet)
+    const finalStatus = this.getFinalTransactionStatus(
+      rawTransactionStatus,
+      lastLedgerPassed,
```

Comment:
do we really need two pieces of information to determine the transaction status? i'm looking at https://xrpl.org/transaction-results.html and it seems to be the case that...

`tec`, `tef`, `tel`, `tem`, and `ter` status codes can NEVER be included in the current ledger; that is, this bit will always be false.

`tes` status codes mean, by definition, that the transaction WAS included in the ledger

does using `lastLedgerPassed` provide more information than expanding TransactionResult to include the rest of the codes? alternatively, it seems like you could get the same logic by having a function called `isSuccessResult`. would this not cover all of the cases we want?

Reply:
another thought: what if we did

```
type FinalTransactionStatus = TransactionStatus.Succeeded | TransactionStatus.Failed | TransactionStatus.Malformed | TransactionStatus.Unknown
```

and then return that in `waitForFinalTransactionOutcome`? then we can abstract away the intermediate pending state

Reply:
got it, per our offline discussion, distributed systems are hard

Reply:
thx for the schoolin' on the ledger, learned a TON today.

my only remaining question is: do we expect to use this function in several places? if `lastLedgerPassed === true`, then `getFinalTransactionStatus` will always return `TransactionStatus.LastLedgerPassed`.

perhaps it would be less ambiguous if we did something like

```
const finalStatus = lastLedgerPassed ? TransactionStatus.LastLedgerPassed : this.getFinalTransactionStatus(rawTransactionStatus)
```

passing `lastLedgerPassed` to `getFinalTransactionStatus` implies that there's some combination of the two where the result can differ. nbd tho

Reply:
Part one of these comments aren't true, but we discussed this offline.